### PR TITLE
90 carousel refactor. 8 - News Carousel

### LINKED ISSUE #8: Implement "Press Release" Carousel
- [ ] requires press release docs and index setup
- [ ] Desktop version
- [ ] Mobile version

Data Source: https://main--moleculardevices--hlxsites.hlx.page/query-index.json?sheet=resource-news
Potential references https://github.com/hlxsites/wknd/blob/main/blocks/carousel/carousel.css#L5-L12 from franklin-wknd

Sample from homepage:

Screenshot 2023-02-21 at 11 44 07

### LINKED ISSUE #90: [refactor] Extract common carousel into a generic implementation
- [ ] Have a generic reusable carousel implementation
- [ ] Have the ability to create new carousel blocks based on it by importing the JS and CSS
- [ ] Make the carousel configurable (e.g. whether to render dots or not)
- [ ] Give the different carousel blocks the ability to render differently their cards, based on their need

### PR DESCRIPTION
**Pull request by andreituicu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #90
Fix #8 

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.page/
- After: https://90-carousel-refactor2--moleculardevices--hlxsites.hlx.page/

- Before: https://main--moleculardevices--hlxsites.hlx.page/drafts/tuicu/carousels
- After: https://90-carousel-refactor2--moleculardevices--hlxsites.hlx.page/drafts/tuicu/carousels

### CODE REVIEW

File: blocks/news-carousel/news-carousel.css [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+main .news-carousel .news-carousel-item-thumb {
+  height: 220px;
+  overflow: hidden;
+}
+
+main .news-carousel .news-carousel-caption {
+  height: calc(100% - 220px);
+  padding: 32px;
+  background-color: #000;
+  color: #fff;
```
[the remark this answers is not in the document]
Comment (andreituicu) [reply blocks/news-carousel/news-carousel.css]: Done in: ecdbc90

File: scripts/scripts.js [hunk 2]
```diff
@@ -18,6 +18,16 @@ import TEMPLATE_LIST from '../templates/config.js';
 const LCP_BLOCKS = []; // add your LCP blocks to the list
 window.hlx.RUM_GENERATION = 'project-1'; // add your RUM generation information here
 
+/**
+ * Converts an excel date to human readable date
+ *
+ * @param {excelDate} the Excel style date
+ * @returns the converted date
+ */
+export function convertDate(excelDate) {
```
[the remark this answers is not in the document]
Comment (andreituicu) [reply scripts/scripts.js]: Removed in: ecdbc90

File: blocks/news-carousel/news-carousel.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  newsCaption.appendChild(newsCaptionText);
+  newsCaption.appendChild(newsCaptionCTA);
+  newsItem.appendChild(newsCaption);
+
+  return newsItem;
+}
+
+export default async function decorate(block) {
+  const newsItems = await ffetch('/query-index.json')
+    .withFetch(fetch)
```
[the remark this answers is not in the document]
Comment (andreituicu) [reply blocks/news-carousel/news-carousel.js]: Done. Thanks!

File: blocks/news-carousel/news-carousel.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import ffetch from '../../scripts/ffetch.js';
+import { createOptimizedPicture } from '../../scripts/lib-franklin.js';
+import createCarousel from '../carousel/carousel.js';
+
+function formatDate(newsDate) {
```
Comment (lydiapuric) [inline blocks/news-carousel/news-carousel.js:5]: I will need this function in news overview. Any chance to put this to scripts.js?

Comment (mhaack) [reply blocks/news-carousel/news-carousel.js:5]: The formatDate? Just move it in a quick PR.

**Pull request merged**